spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.spec.js

```diff
@@ -658,8 +658,18 @@ describe('GA4 core', function () {
     it('populates the timestamp value correctly', function () {
       GOVUK.analyticsGa4.core.sendData({})
       var timestamp = Date.now().toString()
+      var ga4Timestamp = window.dataLayer[0].timestamp
 
-      expect(window.dataLayer[0].timestamp).toEqual(timestamp)
+      // There is sometimes a 1 second delay between the timestamp GA4 will generate and the timestamp this test generates.
+      // Therefore directly comparing timestamps makes this test flaky when the time between both lines of code executing is at least 1 second.
+      // Attempt to resolve this by instead checking the numbers are large enough and are within 5 seconds of each other instead.
+      var exampleTimestamp = 1701234567890
+      var areTimestampsLargeNumbers = timestamp > exampleTimestamp && ga4Timestamp > exampleTimestamp
+
+      var timestampDifference = timestamp - ga4Timestamp
+      var areTimestampsCloseEnough = timestampDifference <= 5 && timestampDifference >= -5
+
+      expect(areTimestampsLargeNumbers && areTimestampsCloseEnough).toEqual(true)
```

Comment:
Good spot - I was sure I've seen this test fail occasionally! Are you aware of the `jasmine.clock()` functionality? Take a look at the option select test for an example - it lets you fast forward through time, so it might allow a more specific test (feel free to split this out into a separate PR if it's easier).

Reply:
Yeah, it's pretty cool - a lot like Timecop, which we use in a few of our other applications.